src/action/auth-mobile.js

```diff
@@ -94,7 +96,11 @@ class AuthAction {
       return;
     }
     await this._setToKeyStore(PIN, newPin);
-    await this._generateWalletPassword();
+    if (this._store.walletUnlocked) {
+      this._nav.goResetPasswordSaved();
+    } else {
+      await this._generateWalletPassword();
+    }
```

Comment:
Mmmmh. Seems like reusing the `SetPassword` (SetPinView) for mobile is creating a bit of added complexity here. It might be more maintainable to create custom views for mobile and use the same routes as on the desktop (`ResetPasswordCurrent`, `ResetPasswordNew`, `ResetPasswordConfirm`, `ResetPasswordSaved`). I know this would have the downside of creating a bunch of redundancy in view code, but would keep action logic simpler and consistent across platforms. I feel like maintaining view code hasn't been the main source of bugs but rather avoiding weird states in the JS logic. What do you think?

Reply:
Does this mean the logic of `pushPinDigit`, etc. would be modified to be more like the previous version of this PR: https://github.com/lightninglabs/lightning-app/pull/939/files#diff-783b94fbd9a84833f5931dd9a59e810dR78?

Reply:
@valentinewallace sorry I didn't see your comment until now. #939 does look more maintainable in the sense that we create custom views for pin reset. So I think we should probably go ahead and review that one instead. Would you mind rebasing on master and opening a new PR? thanks

Reply:
@tanx Sounds good, will do!!